Prepare for release 2.1.5

pt2121 opened the pull request:
This PR prepares the next release `2.1.1`.

* [x] update PTE to the latest commit at [opentransitmap](https://gitlab.com/opentransitmap/public-transport-enabler/-/commits/master) and run `./update-dependency-pinning.sh`
* [x] test the app (automatically using AndroidTest as well as manually), paying special attention to changes since the last release
  * manually tested for London and Paris
  * `searchLocationShowDeparturesTest` has been flaky
  * navitia returns `The region XX-XX doesn't exists"` a lot.
* [x] fix any bugs if necessary
* [] update translations from [Transifex](https://www.transifex.com/) using `tx pull --mode=developer -a` in the root folder (you need proper permissions to do that)
  * I don't have the permission but assumed no new strings?
* [x] ~~add newly supported languages to the two arrays in [`app/src/main/res/values/arrays.xml`](https://github.com/grote/Transportr/blob/master/app/src/main/res/values/arrays.xml#L16)~~
* [x] revise the commits to `master` since the last release and add interesting changes (as well as new languages) to the changelog at [`app/src/main/res/xml/changelog_master.xml`](https://github.com/grote/Transportr/blob/master/app/src/main/res/xml/changelog_master.xml), then run `python3 ./fastlane/generate_changelog.py`
* [x] bump the [`versionCode`](https://github.com/grote/Transportr/blob/master/app/build.gradle#L14) and [`versionName`](https://github.com/grote/Transportr/blob/master/app/build.gradle#L15) in `app/build.gradle`
* [] ~~add a last commit updating all dependencies right after the release~~
  * I will follow up on this.

Altonss commented:
> * `searchLocationShowDeparturesTest` has been flaky

This is probable related to https://github.com/grote/Transportr/issues/727 !

Altonss reviewed (commented):
Thanks for all the work :)

Altonss commented on fastlane/metadata/android/en-US/changelogs/123.txt:
```diff
@@ -0,0 +1,5 @@
+* Add themed launcher Icons.
```
```suggestion
* Support themed adaptive app icon.
```
Idk if this is clearer for end-users?

pt2121 replied:
Got it. I just updated.

Altonss commented on fastlane/metadata/android/en-US/changelogs/123.txt:
```diff
@@ -0,0 +1,5 @@
+* Add themed launcher Icons.
+* Bump minSDK to 21.
```
Would something like a minimum required android version be easier to understand for end-users?

pt2121 replied:
My plan is to let @Echolon take care of the change log in #854 and I'll copy & paste here. :slightly_smiling_face: 
But if that process doesn't work out, I will just gonna make the changes here.

Altonss commented on fastlane/metadata/android/en-US/changelogs/123.txt:
```diff
@@ -0,0 +1,5 @@
+* Add themed launcher Icons.
+* Bump minSDK to 21.
+* Fix bug where calendar and map apps are not found.
+* Update PTE data library - remove VMS, VRS, and PL.
```
See https://github.com/grote/Transportr/discussions/854#discussioncomment-6019116

Altonss commented:
> * navitia returns `The region XX-XX doesn't exists"` a lot in US.

Should some more regions other regions be removed? Like https://github.com/grote/Transportr/issues/838...

Altonss commented:
Maybe we could list in this PR, the associated issues that this PR would close:
- Fixes https://github.com/grote/Transportr/issues/843
- Fixes https://github.com/grote/Transportr/issues/839 
- Solves https://github.com/grote/Transportr/discussions/854

pt2121 commented:
Cool, thanks for the feedback. I'll leave the changelog to @Echolon (and everyone) to make a final decision. I'm good with whatever we'll go with in #854 and just gonna copy and paste it here.

Echolon commented:
> update translations from Transifex using `tx pull --mode=developer -a` in the root folder (you need proper permissions to do that)
>  - I don't have the permission but assumed no new strings?

@ialokim  can you help out with this?

grote commented on app/src/main/java/de/grobox/transportr/networks/TransportNetworks.kt:
```diff
@@ -458,13 +445,6 @@ private val networks = arrayOf(
-                        id = NetworkId.PL,
-                        name = R.string.np_name_pl,
-                        description = R.string.np_desc_pl,
-                        logo = R.drawable.network_pl_logo,
```
If you remove networks, please also remove the associated resources such as strings and drawables. Then the string update for Poland can probably removed from changelog.

pt2121 replied:
Good catch. I removed the related unused resources.

grote commented:
@pt2121 looks great. If you want I can give you access to TRansifex, so you can pull translations.

pt2121 commented:
@grote sounds good, thanks. I have an old transiflex account and my user id is **prt2121**. It's linked to my GitHub account. Let me know if you need any other info to get me the permission.

grote commented:
> I have an old transiflex account and my user id is prt2121. 

transifex can't find this one for some reason.

pt2121 commented:
I created a new account and sent a request to join [the project](https://explore.transifex.com/grobox/transportr/). My new account is [pt2121](https://app.transifex.com/user/profile/pt2121/).
Can you try again when having a chance?

pt2121 commented:
Got the permission. ty.

- I just went ahead and updated the changelog.
- Transifex deprecated the old client so I migrated. Please see [here](https://github.com/transifex/cli#migrating-from-older-versions-of-the-client). 
- Pulled strings and looks like now we have Romanian. Is that expected?

Ready for another round of review

grote commented:
looks good, just odd that there are no other translation changes.

pt2121 commented:
Here is the log:

```
λ tx pull -a
# Getting info about resources

transportr.strings - Done
[##############################] (1 / 1)

# Pulling files

transportr.strings [ca] - Local file is newer than remote, skipping
transportr.strings [cs] - Local file is newer than remote, skipping
transportr.strings [da] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [de] - Local file is newer than remote, skipping
transportr.strings [el] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [eo] - Local file is newer than remote, skipping
transportr.strings [es] - Local file is newer than remote, skipping
transportr.strings [eu] - Local file is newer than remote, skipping
transportr.strings [fa] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [fr] - Local file is newer than remote, skipping
transportr.strings [he] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [hu] - Local file is newer than remote, skipping
transportr.strings [it] - Local file is newer than remote, skipping
transportr.strings [ja] - Local file is newer than remote, skipping
transportr.strings [kab] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [nb] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [nl] - Local file is newer than remote, skipping
transportr.strings [pl] - Local file is newer than remote, skipping
transportr.strings [pt_BR] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [pt_PT] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [ro] - Done
transportr.strings [ru] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [sk] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [sv] - Local file is newer than remote, skipping
transportr.strings [ta] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [tg] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [tr] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [uk] - Minimum translation completion threshold not satisfied, skipping
transportr.strings [zh_TW] - Minimum translation completion threshold not satisfied, skipping
```

I can't tell if this is expected - are the local files actually newer?

pt2121 commented:
Fixed the translation issue. I removed the unused string from the source (en), pushed and then force pulled as you suggested. Also, fixed one translation in Czech where CDATA tag was incorrect.
I think this is good to go now!

grote commented:
Awesome, thanks!

grote commented:
If you want, I can add you to the project's Liberapay team.

pt2121 commented:
Thanks but I don't think I've contributed that much yet - may be in the far future.
You definitely deserve it

grote commented:
They way I see it is that however does the work *now* should get the donations that come in *now*.

pt2121 commented:
I see your point. How about adding me after we get the next release out then? :slightly_smiling_face:
